_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,70 @@
+
+TODO: Why is change lead time a good measure
+$$$ terms
+
+TODO: Why is MTTR a good measure
+$$$ terms
+
+This raises an interesting question to me; if software requires no additional features, and is never in need of repair, is it free of technical debt? Imagine two bodies of work. One has the former traits, along with the reputation for having been terrible to work with by the authors. The other is a more mediocre example of software development, currently in active development and incurring under 5 hours of downtime per year. Which suggests the lesser liability? If I fixate on the numbers in the shadow of our prior definition, I would be led to say the first system has less technical debt. But I guarantee that many would argue in favour of the second. There is an element of potential within technical debt; the absolute zero of which only occurs when the last change to be made is to turn it off.
+
+Waiting for a situation to become untenable is no-ones ideal scenario but we can find ways to mitigate the effects. A development team should be able to reach a consensus as to what are acceptable limits to change-lead-time and MTTR. If these limits are exceeded, it should be acceptable for the development team to focus on reducing them until they are within a tolerable range. Beyer, B. et al. (2016) offer a similar concept called "Error budgets". Specialist engineers are deployed to increase the reliability of critical services that have often been developed by other business units. These engineers assume on-call duties for the service provided that the service meets a minimum standard of stability. If this minimum standard is breached the engineers are within their rights to refuse support until the system can meet it. A development team could enter into a similar contract to relieve technical debt.
+
+As a body of software becomes larger it becomes difficult to maintain. This is aggravated by the inconvenient truth that software engineers are not tethered to their creations. They leave and new blood takes their place. New engineers need to understand the legacy left by the predecessors sans the education of having built the system in the first place. It is unreasonable to expect that a new engineer is going to be as effective as the seasoned veteran. We recognise nobody joins a new company with knowledge of it's internal processes, politics and three-letter-acryonyms, and so we should expect the same of it's software systems. Reducing the amount of learning required to be effective is a suitable means to ensuring that turnover does not effect the mean-time-to-repair and change-lead-time. This also provides the freedom required for specialisation.
```

Comment:
> Reducing the amount of learning required to be effective is a suitable means to ensuring that turnover does not effect the mean-time-to-repair and change-lead-time. This also provides the freedom required for specialisation.

These two points are really impactful. it almost feels like they should be bold and 72px